### PR TITLE
CA trusted fingerprint tests

### PR DESCRIPTION
**Pull request by belimawr**
## What does this PR do?
This PR adds tests for when `ssl.ca_trusted_fingerprint` is set

## Why is it important?
It increases our test coverage and ensures the implementation is correct.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally
1. go to `libbeat/common/transport/tlscommon`
2. run the tests: `go test ./...`

### CODE REVIEW

File: libbeat/common/transport/tlscommon/tls_config_test.go [hunk 1]
```diff
@@ -49,15 +47,15 @@ func TestMakeVerifyServerConnection(t *testing.T) {
 		peerCerts        []*x509.Certificate
 		serverName       string
 		expectedCallback bool
-		expectedError    error
+		expectedError    bool
```
Comment (kvch) [inline libbeat/common/transport/tlscommon/tls_config_test.go:50]: Why did you change this?

Comment (belimawr) [reply libbeat/common/transport/tlscommon/tls_config_test.go:50]: The validation using this field as an error wasn't quite correct, it used be like this:
```go
assert.Error(t, test.expectedError, err)
```

However `assert.Error`'s signature is: `(t TestingT, err error, msgAndArgs ...interface{}) bool`, hence the test was validating that the test case had an error set and using the returned error as a message.

I did try for a wee bit to get one of the other error functions provided by testify to assert the error type, but they didn't quite work, nor I invested much time on it. In the end it was simpler and quicker to just test for error/no error.

Does it make sense? Do we need to be more specific while checking those errors?

Testify can be quite hard to use...

Comment (belimawr) [reply libbeat/common/transport/tlscommon/tls_config_test.go]: I've improved the tests to also validate the error type and message on bc05440788.

**Comment by belimawr** [comment]
rebase onto master, force push

**Review by kvch — APPROVED** [review]
